src/charging-station/ChargingStation.ts

```diff
@@ -738,80 +740,60 @@ export default class ChargingStation {
-      switch (this.getCurrentOutType()) {
-        case CurrentType.AC:
-          limit =
-            matchingChargingProfile.chargingSchedule.chargingRateUnit === ChargingRateUnitType.WATT
-              ? chargingSchedulePeriods[0].limit
-              : ACElectricUtils.powerTotal(
-                  this.getNumberOfPhases(),
-                  this.getVoltageOut(),
-                  chargingSchedulePeriods[0].limit
-                );
-          break;
-        case CurrentType.DC:
-          limit =
-            matchingChargingProfile.chargingSchedule.chargingRateUnit === ChargingRateUnitType.WATT
-              ? chargingSchedulePeriods[0].limit
-              : DCElectricUtils.power(this.getVoltageOut(), chargingSchedulePeriods[0].limit);
+    if (!Utils.isEmptyArray(chargingProfiles)) {
+      const result = this.getCurrentLimitFromOCPPProfiles(chargingProfiles);
+      if (!Utils.isNullOrUndefined(result)) {
+        limit = result.limit;
+        matchingChargingProfile = result.matchingChargingProfile;
+        switch (this.getCurrentOutType()) {
+          case CurrentType.AC:
+            limit =
+              matchingChargingProfile.chargingSchedule.chargingRateUnit ===
+              ChargingRateUnitType.WATT
+                ? limit
+                : ACElectricUtils.powerTotal(this.getNumberOfPhases(), this.getVoltageOut(), limit);
+            break;
+          case CurrentType.DC:
+            limit =
+              matchingChargingProfile.chargingSchedule.chargingRateUnit ===
+              ChargingRateUnitType.WATT
+                ? limit
+                : DCElectricUtils.power(this.getVoltageOut(), limit);
+        }
+
+        const connectorMaximumPower = this.getMaximumPower() / this.stationInfo.powerDivider;
+        if (limit > connectorMaximumPower) {
+          logger.error(
+            `${this.logPrefix()} Charging profile id ${
+              matchingChargingProfile.chargingProfileId
+            } limit is greater than connector id ${connectorId} maximum, dump charging profiles' stack: %j`,
+            this.getConnectorStatus(connectorId).chargingProfiles
+          );
+          limit = connectorMaximumPower;
+        }
       }
     }
-    const connectorMaximumPower = this.getMaximumPower() / this.stationInfo.powerDivider;
-    if (limit > connectorMaximumPower) {
-      logger.error(
-        `${this.logPrefix()} Charging profile id ${
-          matchingChargingProfile.chargingProfileId
-        } limit is greater than connector id ${connectorId} maximum, dump charging profiles' stack: %j`,
-        this.getConnectorStatus(connectorId).chargingProfiles
-      );
-      limit = connectorMaximumPower;
-    }
     return limit;
   }
 
   public setChargingProfile(connectorId: number, cp: ChargingProfile): void {
+    // Add support for connectorID 0
+    if (!Array.isArray(this.getConnectorStatus(connectorId).chargingProfiles)) {
+      this.getConnectorStatus(connectorId).chargingProfiles = [];
+    }
```

Comment:
You need that code because the attribute chargingProfiles is not initialized on connectorId 0? 
If yes, that's an issue in the connectors initialization code and shall be fixed here.